projects/GPGame/game.cpp

```diff
@@ -0,0 +1,15 @@
+#include "game.hpp"
+
+#include <cstdio>
+#include <cassert>
+
+AbstractGame* createGameInstance()
+{
+	return new Game();
+}
+
+void destroyGameInstance(AbstractGame* game)
+{
+	assert(game != nullptr);
+	delete game;
+}
```

Comment:
All this function can be cool inlined

Reply:
No, it can't.
The instance is created in the dll, so it must be destroyed in the dll.
This class must be loaded and called with the exe, so we are forced to use a function ptr to use it.
That's why it can't be inlined.